program/rust/src/processor.rs

```diff
@@ -45,36 +29,26 @@ pub fn process_instruction(
     accounts: &[AccountInfo],
     instruction_data: &[u8],
 ) -> ProgramResult {
-    let cmd_data = load::<cmd_hdr>(instruction_data)?;
-
-    if cmd_data.ver_ != PC_VERSION {
-        return Err(ProgramError::InvalidArgument);
-    }
-
-    match cmd_data
-        .cmd_
-        .try_into()
-        .map_err(|_| OracleError::IntegerCastingError)?
-    {
-        command_t_e_cmd_upd_price | command_t_e_cmd_agg_price => {
-            upd_price(program_id, accounts, instruction_data)
-        }
-        command_t_e_cmd_upd_price_no_fail_on_error => {
+    match load_command_header_checked(instruction_data)? {
+        OracleCommand::InitMapping => init_mapping(program_id, accounts, instruction_data),
+        OracleCommand::AddMapping => add_mapping(program_id, accounts, instruction_data),
+        OracleCommand::AddProduct => add_product(program_id, accounts, instruction_data),
+        OracleCommand::UpdProduct => upd_product(program_id, accounts, instruction_data),
+        OracleCommand::AddPrice => add_price(program_id, accounts, instruction_data),
+        OracleCommand::AddPublisher => add_publisher(program_id, accounts, instruction_data),
+        OracleCommand::DelPublisher => del_publisher(program_id, accounts, instruction_data),
+        OracleCommand::UpdPrice => upd_price(program_id, accounts, instruction_data),
+        OracleCommand::AggPrice => upd_price(program_id, accounts, instruction_data),
+        OracleCommand::InitPrice => init_price(program_id, accounts, instruction_data),
+        OracleCommand::InitTest => Err(OracleError::UnrecognizedInstruction.into()),
+        OracleCommand::UpdTest => Err(OracleError::UnrecognizedInstruction.into()),
+        OracleCommand::SetMinPub => set_min_pub(program_id, accounts, instruction_data),
+        OracleCommand::UpdPriceNoFailOnError => {
             upd_price_no_fail_on_error(program_id, accounts, instruction_data)
         }
-        command_t_e_cmd_resize_price_account => {
+        OracleCommand::ResizePriceAccount => {
             resize_price_account(program_id, accounts, instruction_data)
         }
-        command_t_e_cmd_add_price => add_price(program_id, accounts, instruction_data),
```

Comment:
can you delete the enum in `oracle.h` so no one can depend on it in the C?

Reply:
Some things still depend, just wanted to stage it incrementally